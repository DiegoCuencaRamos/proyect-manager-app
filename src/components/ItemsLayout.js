import React from 'react'
import ListContextProvider from '../contexts/ListContext'
import List from './list/List'
import Calendar from './calendar/Calendar'
import KanbanBoard from './kanban/KanbanBoard'

const ItemsLayout = ({ layout = '' }) => {
    // Render
    if (layout === 'list') {
        return (
            <ListContextProvider>
                <List />
            </ListContextProvider>
        )
    } else if (layout === 'calendar') {
        return <Calendar />
    } else if (layout === 'kanban') {
        return <KanbanBoard />
    }
}

export default ItemsLayout