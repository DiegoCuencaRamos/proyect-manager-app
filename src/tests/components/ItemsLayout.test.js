import React from 'react'
import { shallow } from 'enzyme'
import ItemsLayout from '../../components/ItemsLayout'

test('Should render default ItemsLayout', () => {
    const wrapper = shallow(<ItemsLayout />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render list ItemsLayout', () => {
    const layout = 'list'
    const wrapper = shallow(<ItemsLayout layout={layout} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render calendar ItemsLayout', () => {
    const layout = 'calendar'
    const wrapper = shallow(<ItemsLayout layout={layout} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render kanban ItemsLayout', () => {
    const layout = 'kanban'
    const wrapper = shallow(<ItemsLayout layout={layout} />)
    expect(wrapper).toMatchSnapshot()
})