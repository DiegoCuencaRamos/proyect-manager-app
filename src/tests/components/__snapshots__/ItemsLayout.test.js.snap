// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render calendar ItemsLayout 1`] = `<Calendar />`;

exports[`Should render default ItemsLayout 1`] = `<List />`;

exports[`Should render kanban ItemsLayout 1`] = `<KanbanBoard />`;

exports[`Should render list ItemsLayout 1`] = `<List />`;
